be/src/pipeline/exec/file_scan_operator.cpp

```diff
@@ -62,7 +64,7 @@ void FileScanLocalState::set_scan_ranges(RuntimeState* state,
                                          const std::vector<TScanRangeParams>& scan_ranges) {
     int max_scanners =
             config::doris_scanner_thread_pool_thread_num / state->query_parallel_instance_num();
-    max_scanners = max_scanners == 0 ? 1 : max_scanners;
+    max_scanners = std::max(std::max(max_scanners, state->parallel_scan_max_scanners_count()), 1);
```

Comment:
Do not change this logic in this PR. Maybe in next PR to change the max thread num policy.

Reply:
As mentioned earlier, some scanners may have long tails and require an increase in the number of scanners. Previously, it was not obvious because the blocks were shared, and long tailed scanners can read multiple blocks at once. Currently, only one block can be read, making the problem more serious.